src/query.ts

```diff
@@ -154,10 +157,58 @@ export class Query {
                     marker.fileLine <= toLine
                 );
             }
-        } else throw new Error('Unsupported query format' + value);
+        } else if (value.startsWith('[')) {
+            const propertyQueryMatch = value.match(/\[(.+?):(.*?)\]/);
+            if (!propertyQueryMatch) return false;
+
+            // Quoted property name or value imply exact match; otherwise substring match.
+            const [, propertyNameRaw, propertyQueryRaw] = propertyQueryMatch;
+            const [isExactName, propertyName] = unquote(propertyNameRaw);
+            const [isExactQuery, propertyQuery] = unquote(propertyQueryRaw);
+
+            const fileCache = this.app.metadataCache.getFileCache(marker.file);
+            let propertyValues: string[] = [];
+            if (isExactName) {
+                const property = fileCache.frontmatter?.[propertyName];
+                if (typeof property === 'undefined') propertyValues = property;
+            } else {
+                propertyValues = Object.keys(fileCache.frontmatter)
+                    .filter((k) => k.includes(propertyName))
+                    .map((k) => fileCache.frontmatter[k]);
+            }
+
+            // Allow searching for the existence of a property via a blank value.
+            if (propertyQuery.length === 0) return propertyValues.length > 0;
+
+            const propertyQueryLower = propertyQuery.toLowerCase();
+            return normalizePropertyValues(propertyValues).some((p) =>
+                isExactQuery
+                    ? p === propertyQueryLower
+                    : p.includes(propertyQueryLower)
+            );
+        } else throw new Error('Unsupported query format ' + value);
     }
 }
 
+function normalizePropertyValues(value: unknown): string[] {
+    if (Array.isArray(value)) return value.flatMap(normalizePropertyValues);
+    if (value === null) return ['null'];
+    if (typeof value === 'string') return [value.toLowerCase()];
+    if (['boolean', 'number'].includes(typeof value))
+        return [value.toString().toLowerCase()];
+    throw new Error('Cannot coerce property: ' + value);
+}
+
+// Return whether the given string was quoted, and the unquoted value.
+function unquote(s: string): [boolean, string] {
+    // Match any string, but only capture the first group on balanced quotes.
+    const match = s.match(/^(['"])?(.*)\1$/);
+    if (!match) {
+        throw new Error('Unexpected regex failure when unquoting: ' + s);
+    }
+    return [Boolean(match[1]), match[2]];
+}
+
```

Comment:
Let me know if these belong in `utils.ts` or elsewhere, but I figured I'd put them close to where they're used.

Reply:
Agreed, it's fine here :)